lib/utils/packaging/unarchive.go

```diff
@@ -0,0 +1,150 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package packaging
+
+import (
+	"archive/zip"
+	"io"
+	"os"
+	"path/filepath"
+	"slices"
+	"strings"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/lib/utils"
+)
+
+const (
+	reservedFreeDisk = 10 * 1024 * 1024
+)
+
+// RemoveWithSuffix removes all files in dir that have the provided suffix, except for files named `skipName`
+func RemoveWithSuffix(extractDir, suffix, skipName string) error {
+	err := filepath.Walk(extractDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		if !info.IsDir() {
+			return nil
+		}
+		if skipName == info.Name() {
+			return nil
+		}
```

Comment:
Does this mean `skipName` has to be a directory?

Seems like this method removes all directories in extractDir that have the suffix, except directories named `skipName`. Is that the correct behavior? The godoc says "files"

Reply:
I've added test for this one, and changed logic. Originally this one must be used for cleanup extracted packages when we download and extract new one for another version.